docker/ubuntu-based/prod/vswitch/vppctl

```diff
@@ -13,12 +21,28 @@ then
     exit 1
 fi
 
-# There may be more than one vswitch container if an upgrade/redeploy is happening, so go for the newest one.
-if ! ID=$(docker ps --format '{{.CreatedAt}} {{.Image}} {{.ID}}' | grep -E " $IMAGE@?[^ ]* [^ ]+$" | sort -nr | \
-    head -n 1 | sed -e 's|.* ||g') || [ -z "$ID" ]
-then
-    echo "Error finding a running container derived from the $IMAGE image." >&2
-    exit 1
+ID=$(get_vswitch_container_ID "$IMAGE")
+
+# In K8s 1.10 only Image IDs are displayed by 'docker ps' instead of Image
+# names. If we can't get the vswitch container by its name, try using Image
+# ID, which we get from listing docker images and looking for the latest
+# vswitch image.
+if [ ! "$ID" ] ; then
+    # Just grab the latest (by date) image for now. Will have to be replaced
+    # later with grabbing an image with a specific tag.
+    IMG=$(docker images --format '{{.CreatedAt}};{{.ID}};{{.Repository}}' | grep "$IMAGE" | sort -nr | head -n 1) \
```

Comment:
Grepping for $IMAGE without delimiters could give you a match that you don't want.

Reply:
what delimiters would you suggest? The output line format is very simple, just 3 fields, and none of them is remotely similar to the image name string.